diff -ur Devel.old/Cover/DB.pm Devel/Cover/DB.pm
--- Devel.old/Cover/DB.pm	2008-07-04 12:53:35.000000000 +0200
+++ Devel/Cover/DB.pm	2010-01-29 16:04:27.000000000 +0100
@@ -23,11 +23,11 @@
 my $DB = "cover.12";  # Version 12 of the database.
 
 @Devel::Cover::DB::Criteria =
-    (qw( statement branch path condition subroutine pod time ));
+    (qw( lines statement branch path condition subroutine pod time ));
 @Devel::Cover::DB::Criteria_short =
-    (qw( stmt      bran   path cond      sub        pod time ));
+    (qw( loc   stmt      bran   path cond      sub        pod time ));
 
-# use Data::Dumper; $Data::Dumper::Indent = 1; $Data::Dumper::Sortkeys = 1;
+ use Data::Dumper; $Data::Dumper::Indent = 1; $Data::Dumper::Sortkeys = 1;
 
 sub new
 {
@@ -383,22 +383,28 @@
     my $format = sub
     {
         my ($part, $criterion) = @_;
+        if ($criterion eq "lines") {
+            $criterion = "statement";
+            return $options{$criterion} && exists $part->{$criterion} ?
+                sprintf "%5d", $part->{$criterion}{total} :
+                "n/a"
+        }
         $options{$criterion} && exists $part->{$criterion}
-            ? sprintf "%4.1f", $part->{$criterion}{percentage}
+            ? sprintf "%3.1f", $part->{$criterion}{percentage}
             : "n/a"
     };
 
-    my $fw = 77 - $n * 7;
+    my $fw = 79 - $n * 5;
     $fw = 28 if $fw < 28;
 
     no warnings "uninitialized";
-    my $fmt = "%-${fw}s" . " %6s" x $n . "\n";
-    printf $fmt, "-" x $fw, ("------") x $n;
+    my $fmt = "%-${fw}s" . " %5s" x $n . "\n";
+    printf $fmt, "-" x $fw, ("-----") x $n;
     printf $fmt, "File",
                  map { $self->{all_criteria_short}[$_] }
                  grep { $options{$self->{all_criteria}[$_]} }
                  (0 .. $#{$self->{all_criteria}});
-    printf $fmt, "-" x $fw, ("------") x $n;
+    printf $fmt, "-" x $fw, ("-----") x $n;
 
     my $s = $self->{summary};
     for my $file (grep($_ ne "Total", sort keys %$s), "Total")
@@ -411,7 +417,7 @@
 
     }
 
-    printf $fmt, "-" x $fw, ("------") x $n;
+    printf $fmt, "-" x $fw, ("-----") x $n;
     print "\n\n";
 
     select $oldfh;
@@ -882,6 +888,7 @@
     }
 
     $self->{cover_valid} = 1;
+    $self->{collected}{lines} = undef;
     $self->{cover}
 }
 
diff -ur Devel.old/Cover/Report/Html_minimal.pm Devel/Cover/Report/Html_minimal.pm
--- Devel.old/Cover/Report/Html_minimal.pm	2008-07-04 12:53:35.000000000 +0200
+++ Devel/Cover/Report/Html_minimal.pm	2010-01-29 14:56:59.000000000 +0100
@@ -38,7 +38,15 @@
 
     my $data = $db->{summary}{$file};
     for my $c (@$show) {
-        if (exists $data->{$c}) {
+        if ($c eq "lines") {
+            if (exists $data->{statement}) {
+                $summary{$c} = {
+                    percent => $data->{statement}{total},
+                };
+            } else {
+                $summary{$c} = {percent => 'n/a', ratio => undef, error => undef};
+            }
+        } elsif (exists $data->{$c}) {
             $summary{$c} = {
                 percent => sprintf("%4.1f", $data->{$c}{percentage}),
                 ratio   => sprintf("%d / %d",
@@ -388,7 +396,7 @@
             my $pc = $summary->{$c}{percent};
             my ($class, $popup, $link);
 
-            if ($pc eq 'n/a' || $c eq 'time') {
+            if ($pc eq 'n/a' || $c eq 'time' || $c eq 'lines') {
                 $class = $popup = '';
             }
             else {
@@ -441,6 +449,8 @@
 sub print_file_report {
     my ($db, $fin, $opt) = @_;
 
+    local $opt->{show}{lines};
+
     my $fout = "$opt->{outputdir}/$Filenames{$fin}.html";
     open(my $in,  '<', $fin ) or warn("Can't read file '$fin' [$!]\n"), return;
     open(my $out, '>', $fout) or warn("Can't open file '$fout' [$!]\n"), return;
diff -ur Devel.old/Cover/Report/Html_subtle.pm Devel/Cover/Report/Html_subtle.pm
--- Devel.old/Cover/Report/Html_subtle.pm	2008-07-04 12:53:35.000000000 +0200
+++ Devel/Cover/Report/Html_subtle.pm	2010-01-29 14:56:17.000000000 +0100
@@ -140,6 +140,7 @@
     open(F,'<',  $file) or warn("Unable to open '$file' [$!]\n"), return;
 
     my @lines;
+    local $options->{show}{lines};
     my @showing = grep $options->{show}{$_}, $db->criteria;
     my @headers = map { ($db->all_criteria_short)[$_] }
     grep { $options->{show}{($db->criteria)[$_]} } (0 .. $db->criteria - 1);
Only in Devel/Cover/Report: Html_subtle.pm~
diff -ur Devel.old/Cover.pm Devel/Cover.pm
--- Devel.old/Cover.pm	2008-07-04 12:53:35.000000000 +0200
+++ Devel/Cover.pm	2010-01-29 14:31:47.000000000 +0100
@@ -370,6 +370,9 @@
     $val;
 }
 
+# We cover lines if and only if we cover statements
+*coverage_lines = \&coverage_statement;
+
 sub set_coverage
 {
     set_criteria(cover_names_to_val(@_));
